### PR TITLE
[flutter] assert incorrect types in MethodChannel.invokeMethod

### PR DESCRIPTION
**Pull request by jonahwilliams**
Add a FlutterError to assert for the type error that occurs if a Map or List with a generic type other than dynamic or Object? is given to invokeMethod.

https://github.com/flutter/flutter/issues/80584

### CODE REVIEW

File: packages/flutter/lib/src/services/platform_channel.dart [hunk 1]
```diff
@@ -155,7 +155,40 @@ class MethodChannel {
       }
       throw MissingPluginException('No implementation found for method $method on channel $name');
     }
-    return codec.decodeEnvelope(result) as T?;
+    final Object? resultData = codec.decodeEnvelope(result);
+    assert(() {
+      if (resultData is List<Object?>
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/services/platform_channel.dart:160]: Couldn't think of a better way to do this unfortunately

**Comment by jonahwilliams** [comment]
FYI @xster @Hixie

**Comment by skia-gold** [comment]
Gold has detected about 5 untriaged digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/80609

**Comment by jonahwilliams** [comment]
TIL there are actually tons of places where we use `Map<String, dynamic>` and apparently get away with it. that probably works with the JSON codec

**Comment by skia-gold** [comment]
Gold has detected about 8 untriaged digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/80609

**Comment by jonahwilliams** [comment]
I'm decreasingly sure that this is a good idea

**Comment by jonahwilliams** [comment]
This is sort of weird because in theory I could provide my own codec that reified types differently 🤔

**Comment by jonahwilliams** [comment]
this is actually an awful idea sorry lol!

**Pull request closed without merging**